File: docs/code-examples/entity_path.cpp
// Log a `TextDocument`

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_text_document");
    rec.spawn().exit_on_failure();

    rec.log(
        R"(world/42/escaped\ string\!)",
        rerun::TextDocument("This entity path was escaped manually")
    );
    // TODO: figure this one out
    // rec.log(
    //     {"world", 42, "unescaped string!"},
    //     rerun::TextDocument("This entity path was provided as a list of unescaped strings")
    // );
}

File: docs/code-examples/entity_path.py
import rerun as rr

rr.init("rerun_example_entity_path", spawn=True)

rr.log(r"world/42/escaped\ string\!", rr.TextDocument("This entity path was escaped manually"))
rr.log(
    ["world", 42, "unescaped string!"], rr.TextDocument("This entity path was provided as a list of unescaped strings")
)

assert rr.escape_entity_path_part("my string!") == r"my\ string\!"
assert rr.new_entity_path(["world", 42, "my string!"]) == r"/world/42/my\ string\!"
